saucerswap-faucetbot/main.js

```diff
@@ -0,0 +1,373 @@
+import * as config from "./utils/config.js";
+import fetch from "node-fetch";
+import db from "./configs/db.js";
+import cron from "node-cron";
+import { Client as DiscordClient, Intents } from "discord.js";
+import {
+  AccountBalanceQuery,
+  AccountId,
+  Client,
+  Hbar,
+  PrivateKey,
+  TransferTransaction,
+} from "@hashgraph/sdk";
+
+const faucetBot = new DiscordClient({
+  intents: [
+    Intents.FLAGS.GUILDS,
+    Intents.FLAGS.GUILD_MESSAGES,
+    Intents.FLAGS.DIRECT_MESSAGES,
+  ],
+});
+
+const operatorId = AccountId.fromString(config.TREASURY_ACCOUNT_ID);
+const operatorKey = PrivateKey.fromString(config.TREASURY_PRIVATE_KEY);
+const treasuryId = AccountId.fromString(config.TREASURY_ACCOUNT_ID);
+const treasuryKey = PrivateKey.fromString(config.TREASURY_PRIVATE_KEY);
+const client = Client.forName(config.HEDERA_NETWORK).setOperator(
+  operatorId,
+  operatorKey
+);
+const tokenId = config.SAUCE_TOKEN_ID;
+const accountIdsBeingProcessed = new Set();
+const amount = 0.11;
+const decimals = 6;
+
+faucetBot.on("ready", async () => {
+  console.log(`Logged in as ${faucetBot.user.tag}!`);
+  cron.schedule(
+    "0 */8 * * *",
+    async () => {
+      db.execute(`TRUNCATE TABLE faucet_tracker`);
+      faucetBot.channels.cache
+        .get(config.DISCORD_CHANNEL)
+        .send("Faucet is being reset...!");
+    },
+    {
+      scheduled: true,
+      timezone: "Atlantic/St_Helena",
+    }
+  );
+});
+
+faucetBot.on("messageCreate", async (msg) => {
+  // If the author is the bot, return
+  if (!msg) return;
+  if (msg.author.bot) return;
+
+  // split message into words, distinguish between command & parameters
+  const words = msg.content.split(" ").filter((str) => str.length);
+  let cmd;
+  try {
+    cmd = words[0].toLowerCase();
+  } catch (e) {
+    console.debug("Failed to retrieve command!");
+    return;
+  }
+  const params = words.slice(1);
+  const channel = faucetBot.channels.cache.get(config.DISCORD_CHANNEL);
+  let userMsg = "";
+  try {
+    userMsg = await channel.messages.fetch(msg.id);
+  } catch (e) {
+    console.debug("Failed to retrieve user message!");
+    return;
+  }
+  const time = getTimeTilReset();
+
+  // switch-case on commands
+  switch (cmd) {
+    case "!manual":
+      const response = `Faucet Bot is here to dispense *$SAUCE!*
```

Comment:
make this a configuration option.. for your own sake.. people will get confused if there is SAUCE offered everywhere

Reply:
Resolved in the latest commit.